kernel/src/callback.rs

```diff
@@ -6,41 +6,80 @@ pub struct AppId {
     idx: usize,
 }
 
+/// The kernel can masquerade as an app. IDs >= this value are the kernel.
+/// These IDs are used to identify which kernel container is being accessed.
+const KERNEL_APPID_BOUNDARY: usize = 100;
+
 impl AppId {
     pub fn new(idx: usize) -> AppId {
         AppId { idx: idx }
     }
 
+    pub const fn kernel_new(idx: usize) -> AppId {
+        AppId { idx: idx }
```

Comment:
Oops, those checks were there once upon a refactor. Also checks that new is < boundary now. Good catch.